plugins/woocommerce-admin/client/products/fields/attribute-field/attribute-field.tsx

```diff
@@ -15,29 +20,108 @@ import { closeSmall } from '@wordpress/icons';
 import './attribute-field.scss';
 import AttributeEmptyStateLogo from './attribute-empty-state-logo.svg';
 import { AddAttributeModal } from './add-attribute-modal';
+import { EditAttributeModal } from './edit-attribute-modal';
 import { reorderSortableProductAttributePositions } from './utils';
 
 type AttributeFieldProps = {
 	value: ProductAttribute[];
 	onChange: ( value: ProductAttribute[] ) => void;
+	productId?: number;
+};
+
+export type HydratedAttributeType = Omit< ProductAttribute, 'options' > & {
+	terms: ProductAttributeTerm[];
 };
 
 export const AttributeField: React.FC< AttributeFieldProps > = ( {
 	value,
 	onChange,
+	productId,
 } ) => {
 	const [ showAddAttributeModal, setShowAddAttributeModal ] =
 		useState( false );
+	const [ hydrationComplete, setHydrationComplete ] = useState< boolean >(
+		value ? false : true
+	);
+	const [ hydratedAttributes, setHydratedAttributes ] = useState<
+		HydratedAttributeType[]
+	>( [] );
+	const [ editingAttributeId, setEditingAttributeId ] = useState<
+		null | number
+	>( null );
+
+	const fetchTerms = useCallback(
+		( attributeId: number ) => {
+			return resolveSelect(
+				EXPERIMENTAL_PRODUCT_ATTRIBUTE_TERMS_STORE_NAME
+			)
+				.getProductAttributeTerms< ProductAttributeTerm[] >( {
+					attribute_id: attributeId,
+					product: productId,
+				} )
+				.then(
+					( attributeTerms ) => {
+						return attributeTerms;
+					},
+					( error ) => {
+						return error;
+					}
+				);
+		},
+		[ productId ]
+	);
+
+	useEffect( () => {
```

Comment:
This effect will update the form state when the `hydratedAttributes` are updated.